test/e2e/storage/testsuites/volumeperf.go

```diff
@@ -0,0 +1,327 @@
+			framework.ExpectNoError(err)
+		}
+		ginkgo.By(fmt.Sprintf("Deleting Storage Class %s", l.scName))
+		err := l.cs.StorageV1().StorageClasses().Delete(context.TODO(), l.scName, metav1.DeleteOptions{})
+		framework.ExpectNoError(err)
+		l.testCleanup()
+	})
+
+	ginkgo.It("should provision volumes at scale within performance constraints [Slow] [Serial]", func() {
+		l = &local{
+			cs:      f.ClientSet,
+			ns:      f.Namespace,
+			options: dInfo.PerformanceTestOptions,
+		}
+		l.config, l.testCleanup = driver.PrepareTest(f)
+
+		// Stats for volume provisioning operation
+		// TODO: Add stats for attach, resize and snapshot
+		provisioningStats := &performanceStats{
+			mutex:             &sync.Mutex{},
+			perObjectInterval: make(map[string]*interval),
+			operationMetrics:  &storageframework.Metrics{},
+		}
+		sc := driver.(storageframework.DynamicPVTestDriver).GetDynamicProvisionStorageClass(l.config, pattern.FsType)
+		ginkgo.By(fmt.Sprintf("Creating Storage Class %v", sc))
+		// TODO: Add support for WaitForFirstConsumer volume binding mode
+		if sc.VolumeBindingMode != nil && *sc.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
+			e2eskipper.Skipf("WaitForFirstConsumer binding mode currently not supported for performance tests")
+		}
+		ginkgo.By(fmt.Sprintf("Creating Storage Class %s", sc.Name))
+		sc, err := l.cs.StorageV1().StorageClasses().Create(context.TODO(), sc, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		l.scName = sc.Name
+
+		// Create a controller to watch on PVCs
+		// When all PVCs provisioned by this test are in the Bound state, the controller
+		// sends a signal to the channel
+		controller := newPVCWatch(f, l.options.ProvisioningOptions.Count, provisioningStats)
+		l.stopCh = make(chan struct{})
+		go controller.Run(l.stopCh)
+		waitForProvisionCh = make(chan []*v1.PersistentVolumeClaim)
+
+		ginkgo.By(fmt.Sprintf("Creating %d PVCs of size %s", l.options.ProvisioningOptions.Count, l.options.ProvisioningOptions.VolumeSize))
+		for i := 0; i < l.options.ProvisioningOptions.Count; i++ {
+			pvc := e2epv.MakePersistentVolumeClaim(e2epv.PersistentVolumeClaimConfig{
+				ClaimSize:        l.options.ProvisioningOptions.VolumeSize,
+				StorageClassName: &sc.Name,
+			}, l.ns.Name)
+			pvc, err = l.cs.CoreV1().PersistentVolumeClaims(l.ns.Name).Create(context.TODO(), pvc, metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+			// Store create time for each PVC
+			provisioningStats.mutex.Lock()
+			provisioningStats.perObjectInterval[pvc.Name] = &interval{
+				create: pvc.CreationTimestamp.Time,
+			}
+			provisioningStats.mutex.Unlock()
+			if *sc.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
+				ginkgo.By("creating a pod referring to the claim")
+				var pod *v1.Pod
+				pod, err = e2epod.CreatePod(l.cs, pvc.Namespace, nil /* nodeSelector */, []*v1.PersistentVolumeClaim{pvc}, true /* isPrivileged */, "" /* command */)
```

Comment:
The default max pods per node is 100, so running this with 300 pods is going to fail.

Reply:
Do the prow jobs only run with a single node?

Reply:
Ok I guess the reason why this test is passing is because it's not running with "WaitForFirstConsumer". But I suspect that if you change the hostpath driver test to use it, then setting 300 pods is going to fail because you can only have a max of 100 pods per node.

Reply:
Yes i see what you mean? Do you want me to remove support for WFFC for now and revisit it once i've tested it locally?

Reply:
Actually this test is currently skipping WFFC - https://github.com/kubernetes/kubernetes/pull/94334/files#diff-7f88229dff9847a80f3ff9c3fe105a83c19b0b0ef596186a84cfc29337fbaf88R174

Reply:
I guess since the test is already skipping it then it's fine. It's just confusing because this code will never be executed, so removing it would avoid that confusion.

Reply:
Removed!